File: public/static/templates/nodejs/event/Event.js
forEach: Event
fileName: {{namePascalCase}}.js
path: {{boundedContext.name}}/event
---
const AbstractEvent = require('./AbstractEvent')

class {{namePascalCase}} extends AbstractEvent{
    constructor(){
        super();
        {{#fieldDescriptors}}
        this.{{nameCamelCase}} = null;
        {{/fieldDescriptors}}
    }
}

module.exports = {{namePascalCase}};

File: public/static/templates/spring-boot-mybatis/src/main/java/com/eventstorming/dto/event/Event.java
forEach: Event
fileName: {{namePascalCase}}.java
path: {{boundedContext.name}}/{{{options.packagePath}}}/dto/event
---
package {{options.package}}.dto.event;

import java.util.Date;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Data;

@JsonIgnoreProperties(ignoreUnknown = true)
@Data
public class {{namePascalCase}} extends AbstractEvent {

    {{#fieldDescriptors}}
    private {{className}} {{nameCamelCase}};
    {{/fieldDescriptors}}

    public {{namePascalCase}}(){
        super();
    }
}